src/providers/tpm_provider/key_management.rs

```diff
@@ -203,6 +176,77 @@ impl TpmProvider {
         Ok(psa_import_key::Result {})
     }
 
+    pub(super) fn psa_import_key_internal_rsa_keypair(
+        &self,
+        app_name: ApplicationName,
+        op: psa_import_key::Operation,
+    ) -> Result<psa_import_key::Result> {
+        let key_name = op.key_name;
+        let attributes = op.attributes;
+        let key_triple = KeyTriple::new(app_name, ProviderID::Tpm, key_name);
+        let key_data = op.data;
+
+        let mut store_handle = self
+            .key_info_store
+            .write()
+            .expect("Key store lock poisoned");
+        let mut esapi_context = self
+            .esapi_context
+            .lock()
+            .expect("ESAPI Context lock poisoned");
+
+        let private_key: RSAPrivateKey = picky_asn1_der::from_bytes(key_data.expose_secret())
+            .map_err(|err| {
+                format_error!("Could not deserialise key elements", err);
+                ResponseStatus::PsaErrorInvalidArgument
+            })?;
+
+        // Derive the public key from the keypair.
+        let public_key = RSAPublicKey {
+            modulus: private_key.modulus().clone(),
+            public_exponent: private_key.public_exponent().clone(),
+        };
+
+        // Validate the public and the private key.
+        validate_public_key(&public_key, &attributes)?;
+        validate_private_key(&private_key, &attributes)?;
+
+        let key_prime = private_key.primes()[0].as_unsigned_bytes_be();
+        let public_modulus = private_key.modulus().as_unsigned_bytes_be();
+
+        let (int_bytes, _) = private_key
+            .public_exponent()
+            .split_at(std::mem::size_of::<u32>());
```

Comment:
`split_at` could panic if the public exponent is smaller than 4 bytes but I guess that's fine because of the `validate_` functions above.